editor/components/block-list/invalid-block-warning.js

```diff
@@ -30,16 +30,24 @@ function InvalidBlockWarning( { ignoreInvalid, switchToBlockType } ) {
 		<Warning>
 			<p>{ defaultBlockType && htmlBlockType && sprintf( __(
 				'This block appears to have been modified externally. ' +
-				'Overwrite the external changes or Convert to %s or %s to keep ' +
+				'Overwrite the changes or Convert to %s or %s to keep ' +
 				'your changes.'
 			), defaultBlockType.title, htmlBlockType.title ) }</p>
 			<p>
-				<Button
+				{ block.name === 'core/paragraph' && (
```

Comment:
I wonder if we want to target the block explicitely here or use `getDefaultBlock`.

Maybe a nicer way to handle this is to avoid the message entirely cc #4874

Reply:
Can we do both? I didn't see #4874 that looks good, this code would only be a last resort if it can correct a P block ahead.

Reply:
I think we do get #4874, there's no need to special-case the paragraph block anymore because the most common way to make it invalid will be gone. So this button would become useless.

Reply:
yeh, makes sense.  I already merged this one. So should I simply create a new PR to back out the change?

Reply:
yep let's wait for #4874 to land or maybe it should be included in the PR cc @aduth ?

Reply:
Sounds good to back out the change since handling p-less text as deprecated should fix the fundamental issue.

Separately, we should do an iteration on the invalidation screen, as "Overwrite" is indeed meaningless. We also have the ability to interpret as HTML (as if you had copied and pasted raw HTML which we then convert to blocks) which might be a much better default.

cc @iseulde @jasmussen

Reply:
I like converting to HTML as a primary (recommended) action because it preserves the content (content loss impossible), immediately shows what may be wrong by showing the previously invisible HTML, and the user could continue from there to convert to blocks (we should add a convert to blocks action to the HTML block).

Reply:
We could also give a "convert to blocks" option in here, but that's again another button. I'm unsure how "attempt to fix" would be any different from "convert to blocks". I would also remove "convert to classic" in favour of "edit as HTML block"

Reply:
I'm going to make a PR for this.